user_stories.md

```diff
@@ -0,0 +1,11 @@
+# Пользовательские истории.
+
+1. Как пользователь, я хочу иметь возможность создавать сокращенные ссылки без регистрации на сервисе (возможно, с ограниченным временем работы ссылок), чтобы их было удобнее использовать в текстах, так как они[ccылки] будут со значительно меньшим числом символов.
+2. Как пользователь, я хочу переходить по сокращенной ссылке на страницу, которую указывал создатель этой ссылки.
+3. Как пользователь, я хочу узнать, куда ведет сокращенная ссылка, не переходя по ней явно, чтобы обезопасить себя от возможных вирусов.
+4. Как пользователь, я хочу иметь возможность зарегистрироваться и получить свой аккаунт с уникальным id, чтобы меня можно было отличить от других пользователей.
+5. Как зарегистрированный пользователь, я хочу иметь возможность создавать ссылки с произвольным временем работы, и в любой момент иметь возможность удалить их из системы.
+6. Как зарегистрированный пользователь, я хочу иметь возможность задавать ограничения на символы, используемые в тексте сокращенной ссылки, чтобы удовлетворить требованиям контекста использования.
+7. Как зарегистрированный пользователь, я хочу получить список всех созданных мной ссылок с соответствующими им реальными.
+8. Как зарегистрированный пользователь, я хочу иметь возможность просматривать статистику созданных мной ссылок (количество переходов и т.п.).
```

Comment:
Could you specify the stats you are interested in, please?
Does it imply that there will be no caches between your service and user?

Reply:
Simpliest statistic is the number of clicks on the link, other stats is stats from http headers of query (e.g. browser version), we think in process which of them we implement. I don't understand how cashes are connected to this point.

Reply:
Could you please explain what you mean by "caches between your service and user"?

Statistics (such as the number of clicks on the link, time of link usage or the address from which it was followed) could be useful (for example we gave different promotional links for some bloggers and want to research the interest of the audience of each of them).